src/btree/bt_random.c

```diff
@@ -350,6 +350,37 @@ __random_leaf(WT_CURSOR_BTREE *cbt)
     return (0);
 }
 
+/*
+ * __random_root_inmem_ref --
+ *     Return a random in-mem ref from a root page by applying reservoir sampling.
+ */
+static void
+__random_root_inmem_ref(
+  WT_SESSION_IMPL *session, WT_REF *current, WT_REF **refp, WT_RAND_STATE *rnd)
+{
+    WT_REF *ref, *ref_inmem;
+    uint64_t cnt;
+
+    cnt = 0;
+    ref_inmem = NULL;
+
+    WT_ASSERT(session, __wt_ref_is_root(current));
+
+    WT_STAT_CONN_INCR(session, cache_eviction_random_sample_inmem_root);
+    WT_STAT_DATA_INCR(session, cache_eviction_random_sample_inmem_root);
+
+    WT_INTL_FOREACH_BEGIN (session, current->page, ref)
+        if (ref->state == WT_REF_MEM) {
```

Comment:
There was a merge conflict here
`ref->state` instead of `WT_REF_GET_STATE(ref)`